tools/dwarves/Makefile

```diff
@@ -6,9 +6,9 @@ PKG_NAME:=dwarves
 PKG_VERSION:=1.26
 PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://fedorapeople.org/~acme/dwarves/
-PKG_HASH:=ad4c08339850e404609e2808012580b7e98366d2b91054bb93fe6dca94651fb4
+PKG_HASH:=1d8c9a1c2d42e06cc121a70a39c4f621fd28f15c476ed1b7c7b226f41fdd32df
 
```

Comment:
How about using git and create our own tarballs based on that instead of relying their release? In light of xz fellow dev @dangowrt suggested to use overall more git based sources instead of tarballs, since it's harder to tamper with them. If we change hashes here, might as well switch to git.

```
PKG_SOURCE_PROTO:=git
PKG_SOURCE_URL:=https://github.com/acmel/dwarves.git
PKG_SOURCE_VERSION:=922085e3cff7e976cbb97b2e15c7f866956728ae
PKG_MIRROR_HASH:=xxx
```

I'm not familiar with the performance of bzip2 but from my understanding it's single threaded and didn't see updates in the last 5 ish years.

Reply:
Well, with move to ZSTD this should be feasible without introducing a large performance penalty as XZ compression was rather slow.
It will still be slower than downloading a tarball.

Reply:
I just remembered that for dwarves we can move the source to https://git.kernel.org/pub/scm/devel/pahole/pahole.git/

Which does serve GZ tarballs

Reply:
> How about using git and create our own tarballs based on that instead of relying their release? In light of xz fellow dev @dangowrt suggested to use overall more git based sources instead of tarballs, since it's harder to tamper with them. If we change hashes here, might as well switch to git.

AIUI some members have expressed concerns about doing that, mentioning speed and mirror support among other issues.

> I'm not familiar with the performance of bzip2 but from my understanding it's single threaded and didn't see updates in the last 5 ish years.

Yes, bzip2 has been around for 25 years, it is *very* stable. There are parallel implementations too, should we really need this (although for decompression purposes, I don't know if it makes that much of a difference).

Reply:
I will switch it back to the kernel.org repo, it was switched to the Fedora mirror for 1.23 update.

Reply:
> It will still be slower than downloading a tarball.

Just for clarification, whenever our buildbots download something, they upload the content of `dl/` to sources.openwrt.org, which is CDN'ed.

If the protocol is git, `download.mk` will first try to download the tarball from sources.openwrt.org and only then does a git clone. We're really only cloning the git repository once (via buildbots) and then never again.

> Which does serve GZ tarballs

Cool!

> AIUI some members have expressed concerns about doing that, mentioning speed and mirror support among other issues.

Yes, mirror support is certainly worsened but then again some stuff is only ever available on sources.openwrt.org so it will never go away. Maybe we could look into have more people mirror it? After all it's behind a `fastly`.

> Yes, bzip2 has been around for 25 years, it is very stable. There are parallel implementations too, should we really need this (although for decompression purposes, I don't know if it makes that much of a difference).

While compressing happens only once (see above), decompression happens like all the time so based on [this](https://linuxreviews.org/Comparison_of_Compression_Algorithms#Decompressing_The_Linux_Kernel) the difference is quite big. I know dwarves is not the linux kernel but 8x faster seems like a good argument.

Reply:
@robimarko you're back to using fedorapeople instead of kernel.org, is this what you intended?

Reply:
Yes, as the snapshot feature that provide GZ tarballs on the kernel.org repo does not include git submodules so libbpf is missing and dwarves/pahole wont compile obviously.